skbio/tree/tests/test_tree.py

```diff
@@ -1303,6 +1303,42 @@ def test_shuffle_raises(self):
         with self.assertRaises(MissingNodeError):
             next(self.simple_t.shuffle(names=['x', 'y']))
 
+    def test_cache_leafcount(self):
+        t = TreeNode.read([r'((a,b)e,(c,d)f,g)r;'])
+        t._cache_leafcount()
+        self.assertEquals(t.find('r').leafcount, 5)
```

Comment:
`TreeNode` is a general object though. We should not merge methods into `TreeNode` which are specific for a downstream application. Regardless, the downstream application should use public API methods, and nothing outside of the `TreeNode` namespace should call a `TreeNode` private member method. 

It is the case that these helpers are generally useful though, and it would be nice if they were easily and appropriately accessible. A `property` is one possible means to achieve this. It will be necessary to ensure that a call does not trigger an unnecessary recursion.